packages/cli/src/commands/commerce/catalog/create.ts

```diff
@@ -114,25 +131,64 @@ export default class CatalogCreate extends Command {
     CliUx.ux.action.stop(err ? red.bold('!') : green('✔'));
   }
 
-  private async generateCatalogConfiguration(): Promise<PartialCatalogConfigurationModel> {
+  private async ensureCatalogFields(
+    client: PlatformClient,
+    catalogConfigurationModel: PartialCatalogConfigurationModel,
+    missingFields: FieldModel[]
+  ) {
+    const fieldsToCreate: FieldModel[] = [];
+    const fieldsToUpdate: FieldModel[] = [];
+    const parametrizeField = (field: FieldModel) => ({
+      ...field,
+      multiValueFacet: true,
+      multiValueFacetTokenizers: ';',
+      useCacheForNestedQuery: true,
+    });
+
+    // Get Id fields from catalog configuration
```

Comment:
nitpick, but the comments are a bit superfluous ;) 

The code is pretty self explanatory.